build.rs

```diff
@@ -0,0 +1,37 @@
+extern crate pkg_config;
+
+use std::fs;
+use std::path::Path;
+use std::env;
+
+fn main() {
+    configure_snappy();
+}
+
+fn configure_snappy() {
+    // try pkg_config first
+    if pkg_config::find_library("snappy").is_ok() {
+        return;
+    }
+    if env::var_os("SNAPPY_STATIC").is_some() {
+        println!("cargo:rustc-link-lib=static=snappy");
+        println!("cargo:rustc-flags=-l c++");
+    } else {
+        println!("cargo:rustc-link-lib=dylib=snappy");
+    };
+
+    for f in vec!["/usr/lib","/usr/local/lib"] {
+        if is_file_in("libsnappy.a", Path::new(f)) {
+            println!("cargo:rustc-link-search={}", f);
+        }
+    }
```

Comment:
@xitep, I would also like to not hardcode the paths. However on my system LD_LIBRARY_PATH is empty, and this is likely to be the case for most of our users.

```
echo "foo-$LD_LIBRARY_PATH-baz"
foo--baz
```